File: 프로그래머스/2/60057. 문자열 압축/문자열 압축.js
function solution(s) {
    var answer = s.length;
    const len = s.length;
    
    for (let idx = 1; idx <= Math.floor(len / 2); idx++){
        let compressed = "";
        let cnt = 1;
        let prev = s.substring(0, idx); 
        
        for (let i = idx; i < len; i += idx){
            let curr = s.substring(i, i + idx);
            if (prev == curr) cnt++;
            else {
                compressed += (cnt > 1 ? cnt : "") + prev;
                prev = curr;
                cnt = 1;
            }
        }
        
        compressed += (cnt > 1 ? cnt : "") + prev;
        
        answer = Math.min(answer, compressed.length)
    }

    return answer;
}

File: 프로그래머스/2/60057. 문자열 압축/문자열 압축.java
class Solution {
    public int solution(String s) {
        int answer = s.length();
        int length = s.length();

        for (int u = 1; u <= length / 2; u++) {
            StringBuilder sb = new StringBuilder();
            String prev = s.substring(0, u);
            int cnt = 1;
            
            int c = u;
            for (; c + u <= length; c += u) {
                String curr = s.substring(c, c + u);
                if (curr.equals(prev)) {
                    cnt++;
                } else {
                    if (cnt > 1) {
                        sb.append(cnt);
                    }
                    sb.append(prev);
                    prev = curr;
                    cnt = 1;
                    } 
                }
            if (cnt > 1) {
                sb.append(cnt);
            }
            sb.append(prev);
                        
            if (c < length) {
                sb.append(s.substring(c, length));
            }
            
            answer = Math.min(answer, sb.length());
        }
        
        
        return answer;
    }
}